Lib/threading.py

```diff
@@ -1630,8 +1629,7 @@ def _register_atexit(func, *arg, **kwargs):
     if _SHUTTING_DOWN:
         raise RuntimeError("can't register atexit after shutdown")
 
-    call = functools.partial(func, *arg, **kwargs)
-    _threading_atexits.append(call)
+    _threading_atexits.append(lambda: func(*arg, **kwargs))
```

Comment:
FWIW, `functools.partial()` would be more like:

```python
    _threading_atexits.append(lambda *a, **kw: func(*arg, *a, **kwargs, **kw))
```

However, we control the only caller (`threading._shutdown()`) and it never passes additional args/kwargs, so your change is correct.  I am curious about why we were using `functools.partial()` instead of a lambda in the first place.

Reply:
The relevant code was added in https://github.com/python/cpython/commit/b61b818d916942aad1f8f3e33181801c4a1ed14b. I don't have any more context than that :)